src/components/core/TextManager.js

```diff
@@ -15,15 +17,16 @@ but the architecture allows you to easily stick in your custom Text type.
 More information here :
 https://github.com/felixmariotto/three-mesh-ui/wiki/Using-a-custom-text-type
 
+@template {!Constructor<import('three').Object3D>} T
+@param {T} Base
+
 */
-export default function TextManager( Base = class {} ) {
+export default function TextManager( Base ) {
 
-    return class TextManager extends Base {
```

Comment:
@felixmariotto I should have explained this: this line here improves type safety and intellisense: the `TextManager` class depends on properties that are defined in the `MaterialManager` and `MeshUIComponent` classes.

By extending TextManager from those classes it depends on, the dependencies are clear, and a type system (TypeScript) now knows how to type check, and IDE features like "Go To Definition" know exactly where to go.

This is not a regression, but an improvement.


If you prefer compositional code, that's totally fine! However, this change only _fixes_ the current state of the code base, making it easier to understand for anyone with a type-aware IDE, and does not try to make things compositional.

For composition, we would need to remove use of `extends` and wire things up using containment. I would be happy to explore that, but I'm not sure what you have in mind.

Reply:
Oh! Maybe I'm missing your intent. Do you want to mix `TextManager` with any type of `MaterialManager` or `MeshUIComponet`? Because that is also doable with class mixins, with propert types (not shown in this PR).